### PR TITLE
Started a pull request template

### PR DESCRIPTION
**Pull request by pkerpedjiev**
Started creating a pull request template. Suggestions for what it should include are welcome.

### CODE REVIEW

File: PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+## Description
+
+What was changed in this pull request?
+
+Why is it necessary?
+
+## Checklist
+
+[ ] Unit tests added or updated
```
Comment (flekschas) [inline PULL_REQUEST_TEMPLATE.md]: Shall we turn this into a [checkbox list](https://help.github.com/articles/about-task-lists/) using?
```
- [ ] Unit...
```
E.g.:
- [ ] Unit...

Comment (pkerpedjiev) [reply PULL_REQUEST_TEMPLATE.md]: Yes, that's what I was going for but didn't quite accomplish.

**Comment by pkerpedjiev** [comment]
Uglify -> tensor change is there so that the tests pass.

I added your other changes.

File: PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
+## Description
+
+What was changed in this pull request?
+
+Why is it necessary?
+
```
Comment (alexpreynolds) [inline PULL_REQUEST_TEMPLATE.md]: Perhaps ask for a link to an issue or Slack discussion related to the PR and what it addresses?

**Comment by pkerpedjiev** [comment]
>In the Description section, perhaps it is useful to ask for a link to a Github issue or Slack discussion related to the PR and what it addresses?
Good idea. Not sure how we can link to a Slack discussion but an issue would be helpful.

File: PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+Does it fix an existing issue? If so, please provide a link.
```
Comment (mccalluc) [inline PULL_REQUEST_TEMPLATE.md]:
```suggestion
Fixes #___
```

I think with a string like that github will auto-close the issue, and folks can just delete if there isn't one.

**Pull request merged**